src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -95,6 +100,21 @@ public static boolean shouldBeHearingsInWelshFlag() {
 
     public static List<HearingLocation> getHearingLocations(SscsCaseData caseData,
                                                             ReferenceDataServiceHolder referenceDataServiceHolder) {
+        List<HearingLocation> hearingLocations = getAllHearingLocations(caseData, referenceDataServiceHolder);
+
+        String nextHearingVenueName = caseData.getAdjournCaseNextHearingVenue();
+
+        if (isNotEmpty(nextHearingVenueName) && !HearingsChannelMapping.isPaperCase(caseData)) {
+            return hearingLocations.stream()
+                .filter(location -> location.getLocationId().equals(getVenueID(caseData, nextHearingVenueName)))
+                .collect(Collectors.toList());
+        } else {
+            return hearingLocations;
+        }
```

Comment:
Minor: we genereally dont use elses like this, espically when there is a return.
Its cleaner/easier to read with that block just being inline
```suggestion
        }
        return hearingLocations;
```